File: src/main/java/gestorAplicacion/instalaciones/InstalacionAdultos.java
package gestorAplicacion.instalaciones;

//importar la clase cliente
import gestorAplicacion.servicios.Registro;

import java.util.ArrayList;
import java.io.Serializable;

public class InstalacionAdultos extends Instalacion implements Serializable {

    private static final long serialVersionUID = 1L;
    
    /*private ArrayList<Object> clientesQueSubieron = new ArrayList<>();
    private static ArrayList<Instalacion> instalacionesAdultos = new ArrayList<>();*/
    
    
    public InstalacionAdultos(String nombre, int restriccionEdad){
        super(nombre,restriccionEdad);
    }
    
    @Override
    public String agregarInstalacion(){
        if(!Registro.instalaciones.containsKey(this.getNombre())){
            Registro.instalaciones.put(this.getNombre(), this);
            return "Instalación agregada con éxito";
        }
        return "Ya existe una instalación con ese nombre";
    }
    
    @Override
    public String informacionInstalacion(){
        return "Nombre instalacion: "+getNombre()+
                "\nEdad restricción: 16";
    }
    
    @Override
    public String tipoInstalacion(){
        return "Instalación para adultos";
    }
    
   /* public static void setInstalacionesAdultos(ArrayList<Instalacion> ia){
        instalacionesAdultos = ia;
    }
    
    public static ArrayList<Instalacion> getInstalacionesAdultos(){
        return instalacionesAdultos;
    }*/
    
}

File: practica2/src/gestorAplicacion/instalaciones/InstalacionAdultos.py
from .instalacion import Instalacion
from servicios.registro import Registro

class InstalacionAdultos(Instalacion):

    def __init__(self, nombre):
        Instalacion.__init__(self,nombre, edadRestriccion= "Adulto", costo = 100)
        self._MANTENIMIENTO = 500

    def agregarInstalacion(self):
        """metodo para agregar una nueva instalacion al parque"""
        if not (Registro.instalaciones.containsKey(self.getNombre())):
            Registro.instalaciones.put(self.getNombre(), self)
            return "Instalacion agregada con exito"
            
        return "ya exite una instalacion con ese nombre"

    def getCosto(self):
        return self._costo

    def informacionInstalacion(self):
        """metodo que muestra en pantalla quienes pueden entrar a la instalacion"""
        return f"Esta es una instalacion para mayores de 16"
        
    def setEdadRestriccion (self, edadRestriccion):
        self._edadRestriccion = edadRestriccion
    
    
    def getEdadRestriccion(self):
        return self._edadRestriccion
        
    def setNombre(self, nombre):
        self._nombre = nombre
    
    def getNombre(self):
        return self._nombre

    def sumarUsoAntes(self):
        if self._MANTENIMIENTO > super().getUsosAntes():
            super().sumarUsosAntes()
            return True
        super().solicitarMantenimiento()
        return False

    def __str__(self):
        return f"Nombre: {self.getNombre()}, Restriccion: Mayores,  Costo: 100, Mantenimiento: {super().getNecesitaMantenimiento()}"
